src/components/Card.js

```diff
@@ -4,69 +4,103 @@
     this._link = link;
+    this.isLiked = isLiked;
+    this._likes = likes;
+    this.cardId = _id;
+    this._userId = userId;
     this._cardSelector = cardSelector;
     this._handleCardClick = handleCardClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleLikeClick = handleLikeClick;
   }
 
   _setEventListeners() {
-    // ".cards__like-button";
-    this._cardElement
-      .querySelector(".cards__like-button")
-      .addEventListener("click", () => {
-        this._handleLikeIcon();
-      });
+    this._cardLikeButton.addEventListener("click", () =>
+      this._handleLikeClick(this)
+    );
+    this._cardDeleteButton.addEventListener("click", () => {
+      this._handleDeleteClick(this);
+    });
+    this._cardImage.addEventListener("click", () => {
+      this._handleCardClick(this._getData());
+    });
+  }
 
-    // ".cards__delete-button"
-    this._cardElement
-      .querySelector(".cards__delete-button")
-      .addEventListener("click", () => {
-        this._handleDeleteCard();
-      });
+  _renderLikes() {
+    if (this.isLiked) {
+      this._cardLikeButton.classList.add("cards__like-button_active");
+    } else {
+      this._cardLikeButton.classList.remove("cards__like-button_active");
+    }
+  }
 
-    //#preview-modal
-    this._cardElement
-      .querySelector(".cards__image")
-      .addEventListener("click", () => {
-        this._handlePreviewDisplay();
-      });
+  updateLikes(isLiked) {
+    this.isLiked = isLiked;
+    this._renderLikes();
   }
 
-  _handleLikeIcon() {
-    this._cardElement
-      .querySelector(".cards__like-button")
-      .classList.toggle("cards__like-button_active");
+  _showCardLikes() {
+    this._cardLikeCounter.textContent = this._likes.length;
+    if (this.cardIsLiked()) {
+      this._cardLikeButton.classList.add("cards__like-button_active");
+    } else {
+      this._cardLikeButton.classList.remove("cards__like-button_active");
+    }
   }
 
-  _handleDeleteCard() {
+  handleDeleteCard() {
     this._cardElement.remove();
     this._cardElement = null;
   }
 
-  _handlePreviewDisplay() {
-    this._handleCardClick(this._name, this._link);
-  }
-
   _getTemplate() {
     return document
       .querySelector(this._cardSelector)
       .content.querySelector(".cards__content")
       .cloneNode(true);
   }
 
+  _getData() {
+    return {
+      name: this._name,
+      link: this._link,
+    };
+  }
+
   getView() {
     this._cardElement = this._getTemplate();
 
-    const cardImage = this._cardElement.querySelector(".cards__image");
-    const cardTitle = this._cardElement.querySelector(".cards__title");
+    this._cardImage = this._cardElement.querySelector(".cards__image");
+
+    this._cardLikeButton = this._cardElement.querySelector(
+      ".cards__like-button"
+    );
+    this._cardDeleteButton = this._cardElement.querySelector(
+      ".cards__delete-button"
+    );
+    this._cardLikeCounter = this._cardElement.querySelector(".cards__counter");
```

Comment:
Good job that you've found these variables only once here and defined them as **class fields**